scripts/exglobal_stage_ic.sh

```diff
@@ -8,6 +8,28 @@ GDATE=$(date -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
 gPDY="${GDATE:0:8}"
 gcyc="${GDATE:8:2}"
 
+# Determine NMEM_ENS based on the configuration file (e.g., config.base)
+# Assuming NMEM_ENS is defined in config.base as an integer
+
+config_file="${EXPDIR}/config.base"
+
+if [[ -e "${config_file}" ]]; then
+    NMEM_ENS=$(grep "NMEM_ENS" "${config_file}" | awk -F= '{print $2}' || true)
+else
+    echo "Error: ${config_file} does not exist or is not accessible." >&2
+    exit 1
+fi
+
+member_dirs=()
+
+# Populate the member_dirs array based on the value of NMEM_ENS
+for ((i = 0; i < "${NMEM_ENS}"; i++)); do
+    member_dirs+=("mem$(printf "%03d" "${i}")")
+done
```

Comment:
See associated comment about not using big if-blocks or a command-line argument.
```suggestion
if [[ "${RUN}" == "GEFS" ]]; then
  # Populate the member_dirs array based on the value of NMEM_ENS
  for ((i = 0; i < "${NMEM_ENS}"; i++)); do
      member_dirs+=("mem$(printf "%03d" "${i}")")
  done
else
  member_dirs+=("")
fi
```

Reply:
Updated